README.md
## Overview
folioGallery is an easy to use gallery written by Harry G of foliopages.com.  Just drop it on your PHP web server and edit ``foliogallery.php`` to point the ``$mainFolder`` variable to the location of your pictures you want hosted.  If you are like me, this makes it super easy to turn your pile of home pictures into a browsable website in just a few minutes!

![folioGallery](/screenshot.png)


## DISCLAIMER

I did not create this!  [See the original project from foliopages.com here](http://www.foliopages.com/php-jquery-ajax-photo-gallery-no-database).


## Improvements
### Scale
When you have hundreds (or thousands) of page numbers, they would line up off the window and be unclickable.  This makes sure that does not happen anymore.
### Pagination
Pages are now shown as numbers instead of circle images



changelog.txt
version 1.3.1
- Added improvements to very large albums

version 1.3
- Added option to display random or static album thumbnails
- Modified thumbnail resizing function. Now it resizes images only if they are wider than specified
- Renamed some php vriables in code
- Sort by name is not case sensitive now
- Cleaned up code and fixed some other bugs
- Changed appearance of pagination   
- Made other minor modifications

version 1.2
- Renamed the gallery to folioGallery
- Added option for displaying captions in album files
- Added png and gif automatic thumbnail creation
- Changed thumbnail styling
- Modified the CSS 
- Made other minor modifications

Version 1.1
- Correct number of images are now displayed in colorbox modal window.
- Correct number of images are now looped in colorbox modal window. 
- Made minor changes to php file.


index.php
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>
  <?php 
    if (isset($_GET['album'])) {
	  echo $_GET['album'];
	} else {
	  echo 'Photo Gallery';
	}
  ?>
</title>

<!-- start gallery header --> 
<link rel="stylesheet" type="text/css" href="folio-gallery.css" />
<script type="text/javascript" src="js/jquery-1.9.1.min.js"></script>

<link rel="stylesheet" type="text/css" href="colorbox/colorbox.css" />
<!--<link rel="stylesheet" type="text/css" href="fancybox/fancybox.css" />-->
<script type="text/javascript" src="colorbox/jquery.colorbox-min.js"></script>
<!--<script type="text/javascript" src="fancybox/jquery.fancybox-1.3.1.min.js"></script>-->
<script type="text/javascript">
$(document).ready(function() {	
	
	// colorbox settings
	$(".albumpix").colorbox({rel:'albumpix', transition:'none', scalePhotos:'true', maxWidth:'950px', maxHeight:'720px'});
	
	// fancy box settings
	/*
	$("a.albumpix").fancybox({
		'autoScale	'		: true, 
		'hideOnOverlayClick': true,
		'hideOnContentClick': true
	});
	*/
});
</script>
<!-- end gallery header -->
</head>
<body>

<p>&nbsp;</p>

<div class="gallery">  
  <?php include "folio-gallery.php"; ?>
</div>   

</body>
</html>


folio-gallery.php
<?php 
// error_reporting (E_ALL ^ E_NOTICE);
// photo gallery settings
$mainFolder    = 'albums';   // folder where your albums are located - relative to root
$albumsPerPage = '6';       // number of albums per page
$itemsPerPage  = '9';       // number of images per page    
$thumb_width   = '150';      // width of thumbnails
//$thumb_height  = '85';       // height of thumbnails
$extensions    = array(".jpg",".jpeg",".png",".gif",".JPG","JPEG",".PNG",".GIF"); // allowed extensions in photo gallery
$ignore  = array('.', '..', 'thumbs'); // list of folders to ignore from album listing


// create thumbnails from images
function make_thumb($folder,$src,$dest,$thumb_width) {

		// ensure thumbnail folder exits	   
		if (!is_dir($folder.'/thumbs')) {
				mkdir($folder.'/thumbs');
				chmod($folder.'/thumbs', 0777);
		}

	$source_image = imagecreatefromjpeg($folder.'/'.$src);
	$width = imagesx($source_image);
	$height = imagesy($source_image);
	
	$thumb_height = floor($height*($thumb_width/$width));
	
	$virtual_image = imagecreatetruecolor($thumb_width,$thumb_height);
	
	imagecopyresampled($virtual_image,$source_image,0,0,0,0,$thumb_width,$thumb_height,$width,$height);
	
	imagejpeg($virtual_image,$dest,100);
	
}

// display pagination
function print_pagination($numPages,$urlVars,$currentPage) {
        
   if ($numPages > 1) {
      
	   echo '<b>Page '. $currentPage .' of '. $numPages . '</b>';
	   echo '<br />';
   
       if ($currentPage > 1) {
	       $prevPage = $currentPage - 1;
	       echo '<a href="?'. $urlVars .'p='. $prevPage.'">&laquo;&laquo; Last Page </a> ';
	   }	   
	   if ($currentPage != $numPages) {
           $nextPage = $currentPage + 1;	
		   echo ' <a href="?'. $urlVars .'p='. $nextPage.'"> Next Page &raquo;&raquo;</a>';
	   }	  	 

		echo "<div class='paginate-pages'>";

	   for( $e=0; $e < $numPages; $e++ ) {
           $p = $e + 1;
       
	       if ($p == $currentPage) {	    
		       $class = 'current-paginate';
	       } else {
	           $class = 'paginate';
	       } 
	       

		       echo '<a class="'. $class .'" href="?'. $urlVars .'p='. $p .'">'. $p .'</a> ';
		  	  
	   }

		echo "</div>";
	   
   
   }

}


// if an album is not chosen, find a list of folders in the directory and select a random picture from each
if (!isset($_GET['album'])) {

    // display list of albums
    $folders = scandir($mainFolder, 0);
		  
	$albums = array();
	$captions = array();
	$random_pics = array();
	 
	// inspect each folder in directory 
    foreach($folders as $folder) {
        
		// if the folder is not in the ignore list 
	    if(!in_array($folder, $ignore)) {    
		
			// album name
			$album['name'] = $folder;

			// album caption 
			$album['caption'] = substr($album['name'],0,20);

			// loop through each item in directory and scan for a cover image
			$dir = opendir("$mainFolder/" . $album['name']);
			while ( $entry = readdir($dir) ) {
				if ( $entry != '.' && $entry != '..' ) { //Skips over . and ..
		   			$extension = strstr($entry, '.');
		   			if ( in_array($extension, $extensions)) { // skip items not in our specified supported extensions
						$album['coverImage'] = $entry;
						break;
					}
				}
			}


			// push our newly created album object into an array of all albums	
		   array_push( $albums, $album );

		  
		 } // end if folder is not in ignore array
		  
	 } // end foreach folder in dir

 
	// if there were 0 albums found 
     if( count($albums) === 0 ) {
  
        echo "There are currently no albums.  Add some to $mainFolder and they will appear here.";
  
     } else {
 
		// calculate number of albums per page  
		$numPages = ceil( count($albums) / $albumsPerPage );

		// if a page was passed in the url
        if(isset($_GET['p'])) {
     
				// fetch the requested page id 
				$currentPage = $_GET['p'];
				// if page number is above the maximum, seve the last posible page
				if($currentPage > $numPages) {
						$currentPage = $numPages;
				}

		} else {
				// if no page was specified, choose page 1
				$currentPage = 1;
		} 


		// calculate the first album to show  for the selected page
         $start = ( $currentPage * $albumsPerPage ) - $albumsPerPage;
	 

		// display titlebar  html
	     echo '<div class="titlebar">
                 <div class="float-left"><span class="title">Photo Gallery</span> - Albums</div>
			     <div class="float-right">'.count($albums).' albums</div>
              </div>';
	  
         echo '<div class="clear"></div>';
	  	  		

		// slice our album list depending on the page we are on
		$albums = array_slice($albums,$start,$albumsPerPage);	

		// loop through each album 
	     foreach($albums as $album) {
	 
				// find the filesystem location of the thumbnail we will create 
				$thumbDest = $mainFolder . '/' . $album['name'] . '/thumbs/' . $album['coverImage'];
		     	make_thumb( $mainFolder . '/' . $album['name'] . '/', $album['coverImage'], $thumbDest, $thumb_width); 

				// prepend with a slash to get the URL for this thumbnail

			 		 			 
			    echo '<div class="thumb-album shadow">
				        
						<div class="thumb-wrapper">
						   <a href="'.$_SERVER['PHP_SELF'].'?album='. urlencode($album['name']) .'">
			                 <img src="/'.$thumbDest.'" width="'.$thumb_width.'" alt="" />
						   </a>	
					    </div>
						
						<div class="p5"></div>
					    
						<a href="'.$_SERVER['PHP_SELF'].'?album='. urlencode($album['name']) .'">
							<span class="caption">'. $album['caption'] .'</span>
						</a>
		            
					  </div>';
				  
	      }
	  
	      echo '<div class="clear"></div>';
 

		// display page pagination 
          echo '<div align="center" class="paginate-wrapper">';
        	 
                 $urlVars = "";
                 print_pagination($numPages,$urlVars,$currentPage);
  
          echo '</div>';	   
   
     }
  
 
// If an album is chosen
} else {

     // display photos in album
     $src_folder = $mainFolder.'/'.$_GET['album'];
     $src_files  = scandir($src_folder);

     $files = array();


	 // filter all files by file extension
     foreach($src_files as $file) {
        
		$ext = strrchr($file, '.');
        if(in_array($ext, $extensions)) {
          
		   array_push( $files, $file );
        
		 }
      
	  }



     // slice array thumbs down to the limited ones that should be thumbnails and generate them
	 $currentPage = $_GET['p'];
	 if($currentPage <= 1){
	 	$startThumb = 0;
	 } else {
	 	$startThumb = ($currentPage - 1) * $itemsPerPage;
		$startThumb = $startThumb;
	 }
	 $makeIntoThumbs = array_slice($files,$startThumb,$itemsPerPage);
     foreach($makeIntoThumbs as $file) {

			 $thumb = $src_folder.'/thumbs/'.$file;
			 if (!file_exists($thumb)) {
					 make_thumb($src_folder,$file,$thumb,$thumb_width); 
			 } else {
			 }
	}
 

   if ( count($files) == 0 ) {

      echo 'There are no photos in this album!';
   
   } else {
   
      $numPages = ceil( count($files) / $itemsPerPage );

      if(isset($_GET['p'])) {
      
	     $currentPage = $_GET['p'];
         if($currentPage > $numPages) {
            $currentPage = $numPages;
         }

      } else {
         $currentPage=1;
      } 

   $start = ( $currentPage * $itemsPerPage ) - $itemsPerPage;

   echo '<div class="titlebar">
           <div class="float-left"><span class="title">'. $_GET['album'] .'</span> - <a href="'.$_SERVER['PHP_SELF'].'">View All Albums</a></div>
           <div class="float-right">'.count($files).' images</div>
         </div>';	  
   echo '<div class="clear"></div>';


   for( $i=$start; $i<$start + $itemsPerPage; $i++ ) {
		  
		  if( isset($files[$i]) && is_file( $src_folder .'/'. $files[$i] ) ) { 
	   
	        echo '<div class="thumb shadow">
	                <div class="thumb-wrapper">
					<a href="'. $src_folder .'/'. $files[$i] .'" class="albumpix" rel="albumpix">
				      <img src="'. $src_folder .'/thumbs/'. $files[$i] .'" width="'.$thumb_width.'" alt="" />
				    </a>
					</div>  
			      </div>'; 
      
	      } else {
		  
		    if( isset($files[$i]) ) {
			  echo $files[$i];
		    }
		   
		  }
     
    }
	   

     echo '<div class="clear"></div>';
  
     echo '<div align="center" class="paginate-wrapper">';
        	 
        $urlVars = "album=".urlencode($_GET['album'])."&amp;";
        print_pagination($numPages,$urlVars,$currentPage);
  
     echo '</div>';
	 
	 
   } // end else	 

}
?>


foliogallery-demo.php
<!DOCTYPE html>
<head>
<title>Image Gallery By FolioPages.com</title>
<style type="text/css">
body {
background:#eee;
margin:0;
padding:0;
font:12px arial, Helvetica, sans-serif;
color:#222;
}
</style>
<link type="text/css" rel="stylesheet" href="foliogallery/foliogallery.css" />
<link type="text/css" rel="stylesheet" href="colorbox/colorbox.css" />
<script type="text/javascript" src="foliogallery/jquery.1.11.js"></script>
<script type="text/javascript" src="foliogallery/foliogallery.js"></script>
<script type="text/javascript" src="colorbox/jquery.colorbox-min.js"></script>
</head>
<body>

<br />
<br />

<!-- display images located in "Lake Tahoe" subfolder and paginate in 3 item blocks -->
<div id="folioGallery2" class="folioGallery" title="Lake Tahoe"><div class="numPerPage" title="3"></div></div>

<br />
<br /> 

<!-- display images located in "Las Vegas Strip" subfolder and paginate in 3 item blocks -->
<div id="folioGallery3" class="folioGallery" title="Las Vegas Strip"><div class="numPerPage" title="3"></div></div>

<br />
<br />

<!-- display list of folders in "albums" folder and paginate in 6 item blocks -->
<div id="folioGallery" class="folioGallery"><div class="numPerPage" title="3"></div></div>

<br />
<br />

<div align="center">folioGallery - Installation and instructions @ <a href="http://foliopages.com/php-jquery-ajax-photo-gallery-no-database">FolioPages.com</a></div>

<br />
<br />


</body>
</html>

foliogallery/foliogallery.js
/*!
	folioGallery v1.3 - 2014-05-01
	(c) 2014 Harry Ghazanian - foliopages.com/php-jquery-ajax-photo-gallery-no-database
	This content is released under the http://www.opensource.org/licenses/mit-license.php MIT License.
*/
$(function() {		
				
	var folioGalleryDir = './'; // foliogallery folder relative path - absolute path like http://my_website.com/foliogallery may not work
		
	// find divs with class folioGallery and load album in it based on its id
	$('.folioGallery').each(function() {
		
		var targetDiv = (this.id); // id of div to load albums
		var numPerPage = $('#'+targetDiv+' div.numPerPage').prop('title');
								
		if(targetDiv=='folioGallery') {
			var fullAlbum = 1;
			var showAlb = ''; // empty will show full gallery
		} else {
			var fullAlbum = 0;
			var showAlb = $('#'+targetDiv).prop('title'); // title attribute of div - same as album folder
		}
		
		loadGallery(folioGalleryDir,targetDiv,showAlb,numPerPage,1,fullAlbum); // inital load
				
		// in gallery view, load album when thumb is clicked
		$(this).on('click', 'a.showAlb', function() {	
			var showAlb = $(this).prop('rel');
			loadGallery(folioGalleryDir,targetDiv,showAlb,numPerPage,1,fullAlbum);
			return false;
		});	
				
		// paginate albums and pics
		$(this).on('click', 'a.pag', function() {	
			var showAlb = $(this).prop('rel');
			var pageNum = $(this).prop('rev');
			loadGallery(folioGalleryDir,targetDiv,showAlb,numPerPage,pageNum,fullAlbum);
			return false;
		});
		
		// refresh div content
		$(this).on('click', 'a.refresh', function() {
		   loadGallery(folioGalleryDir,targetDiv,'',numPerPage,1,fullAlbum);
		   return false;
		});
		
				
		// colorbox
		$(this).on('click', 'a.albumpix', function() {$('#'+targetDiv+' .albumpix').colorbox({rel:targetDiv, maxWidth:'98%', maxHeight:'98%', slideshow:true, slideshowSpeed:3500, slideshowAuto:false}); });

	});
		
});

function loadGallery(folioGallerydir,targetdiv,album,numperpage,pagenum,fullalbum) {                    
	$.ajax
	({
		type: 'POST',
		url: folioGallerydir+'/foliogallery.php?'+album+'&p='+pagenum,
		data: {
			album: album,
			numperpage: numperpage,
			pagenum: pagenum,
			fullalbum: fullalbum
		},
		cache: false,
		success: function(msg)
		{
			$('#'+targetdiv).html(msg).hide().show();
		}
	});
	return false;
}

foliogallery.php
<!--
	folioGallery v1.3 - 2014-05-01
	(c) 2014 Harry Ghazanian - foliopages.com/php-jquery-ajax-photo-gallery-no-database
	This content is released under the http://www.opensource.org/licenses/mit-license.php MIT License.
-->
<?php
/***** gallery settings *****/
$mainFolder          = 'albums'; // main folder that holds albums - this folder resides on root directory of your domain
$album_page_url      = $_SERVER['PHP_SELF']; // url of page where gallery/albums are located 
$no_thumb            = 'foliogallery/noimg.png';  // show this when no thumbnail exists 
$extensions          = array("jpg","png","gif","JPG","PNG","GIF"); // allowed extensions in photo gallery 
$itemsPerPage        = '12';    // number of images per page if not already specified in ajax mode 
$thumb_width         = '150';   // width of thumbnails in pixels
$sort_albums_by_date = FALSE;    // TRUE will sort albums by upload date, FALSE will sort albums by name 
$sort_images_by_date = TRUE;    // TRUE will sort thumbs by creation date, FALSE will sort images by name 
$random_thumbs       = TRUE;    // TRUE will display random thumbnails, FALSE will display the first image from thumbs folders
$show_captions       = TRUE;    // TRUE will display file names as captions on thumbs inside albums, FALSE will display no captions
$num_captions_chars  = '20';    // number of characters displayed in album and thumb captions
/***** end gallery settings *****/

$numPerPage = (!empty($_REQUEST['numperpage']) ? (int)$_REQUEST['numperpage'] : $itemsPerPage);
$fullAlbum  = (!empty($_REQUEST['fullalbum']) ? 1 : 0);
 
// function to create thumbnails from images
function make_thumb($folder,$file,$dest,$thumb_width) {

	$ext = strtolower(pathinfo($file, PATHINFO_EXTENSION));
	
	switch($ext)
	{
		case "jpg":
		$source_image = imagecreatefromjpeg($folder.'/'.$file);
		break;
		
		case "jpeg":
		$source_image = imagecreatefromjpeg($folder.'/'.$file);
		break;
		
		case "png":
		$source_image = imagecreatefrompng($folder.'/'.$file);
		break;
		
		case "gif":
		$source_image = imagecreatefromgif($folder.'/'.$file);
		break;
	}	
	
	$width = imagesx($source_image);
	$height = imagesy($source_image);
	
	if($width < $thumb_width) // if original image is smaller don't resize it
	{
		$thumb_width = $width;
		$thumb_height = $height;
	}
	else
	{
		$thumb_height = floor($height*($thumb_width/$width));
	}
	
	$virtual_image = imagecreatetruecolor($thumb_width,$thumb_height);
	
	if($ext == "gif" or $ext == "png") // preserve transparency
	{
		imagecolortransparent($virtual_image, imagecolorallocatealpha($virtual_image, 0, 0, 0, 127));
		imagealphablending($virtual_image, false);
		imagesavealpha($virtual_image, true);
    }
	
	imagecopyresampled($virtual_image,$source_image,0,0,0,0,$thumb_width,$thumb_height,$width,$height);
	
	switch($ext)
	{
	    case 'jpg': imagejpeg($virtual_image, $dest,80); break;
		case 'jpeg': imagejpeg($virtual_image, $dest,80); break;
		case 'gif': imagegif($virtual_image, $dest); break;
		case 'png': imagepng($virtual_image, $dest); break;
    }

	imagedestroy($virtual_image); 
	imagedestroy($source_image);
	
}

// return array sorted by date or name
function sort_array(&$array,$dir,$sort_by_date) { // array argument must be passed as reference
	
	if($sort_by_date)
	{
		foreach ($array as $key=>$item) 
		{
			$stat_items = stat($dir .'/'. $item);
			$item_time[$key] = $stat_items['ctime'];
		}
		return array_multisort($item_time, SORT_DESC, $array); 
	}	
	else
	{
		return usort($array, 'strnatcasecmp');
	}	

}

// display pagination
function paginate_array($numPages,$urlVars,$alb,$currentPage) {
        
   $html = '';
   
   if ($numPages > 1) 
   {
       if ($currentPage > 1)
	   {
	   	   $html .=  "<div id='nextPage'>";
	       $prevPage = $currentPage - 1;
	       $html .= ' <a class="pag prev" rel="'.$alb.'" rev="'.$prevPage.'" href="?'.$urlVars.'p='.$prevPage.'">Last Page</a> ';
	   	   $html .=  '</div>';
	   }	   
	   
	   if ($currentPage != $numPages)
	   {
	   	   $html .=  "<div id='prevPage'>";
           $nextPage = $currentPage + 1;	
		   $html .= ' <a class="pag next" rel="'.$alb.'" rev="'.$nextPage.'" href="?'.$urlVars.'p='.$nextPage.'">Next Page</a>';
	   	   $html .=  '</div>';
	   }	  	 

	   for( $i=0; $i < $numPages; $i++ )
	   {
           $p = $i + 1;
		   $class = ($p==$currentPage ? 'current-paginate' : 'paginate'); 
		   $html .= '<a rel="'.$alb.'" rev="'.$p.'" class="'.$class.' pag" href="?'.$urlVars.'p='.$p.'">'.$p.'</a>';	  
	   }
   
   }
   
   return $html;

}
?>

<div class="fg">

<?php
if (empty($_REQUEST['album'])) // if no album requested, show all albums
{		
	
	$albums = array_diff(scandir($mainFolder), array('..', '.'));
	$numAlbums = count($albums);
	 
	if($numAlbums == 0) 
	{ ?>
		
		<div class="titlebar"><p>There are currently no albums</p></div>
    
	<?php
	}
	else
	{
		sort_array($albums,$mainFolder,$sort_albums_by_date); // rearrange array either by date or name
		$numPages = ceil( $numAlbums / $numPerPage );
		
		if(isset($_REQUEST['p']))
		 {
		 	$currentPage = ((int)$_REQUEST['p'] > $numPages ? $numPages : (int)$_REQUEST['p']); 
         } 
		 else 
		 {
		 	$currentPage=1;
         }
		
		$start = ($currentPage * $numPerPage) - $numPerPage; ?>
	     
		<div class="p10-lr">
        	<span class="title">Photo Gallery</span> - <?php echo $numAlbums; ?> albums
        </div>
	  
        <div class="clear"></div>
	  	 
		<?php 			 
	    for( $i=$start; $i<$start + $numPerPage; $i++ )
		{
	  						
			if(isset($albums[$i])) 
			{  
				$thumb_pool = glob($mainFolder.'/'.$albums[$i].'/thumbs/*{.'.implode(",", $extensions).'}', GLOB_BRACE);
				
				if (count($thumb_pool) == 0)
				{ 
					$album_thumb = $no_thumb;
				}
				else
				{	
					$album_thumb = ($random_thumbs ? $thumb_pool[array_rand($thumb_pool)] : $thumb_pool[0]); // display a random thumb or the 1st thumb					
				} ?>
			 		 			 
				<div class="thumb-wrapper">
					<div class="thumb">
					   <a class="showAlb" rel="<?php echo $albums[$i]; ?>" href="<?php echo $_SERVER['PHP_SELF']; ?>?album=<?php echo urlencode($albums[$i]); ?>">
					     <img src="<?php echo $album_thumb; ?>" alt="<?php echo $albums[$i]; ?>" /> 
					   </a>	
					</div>
					<div class="caption"><?php echo substr($albums[$i],0,$num_captions_chars); ?></div>
				</div>
	
			<?php
			}
		
		}
		?>
	      
		 <div class="clear"></div>
  
         <div align="center" class="paginate-wrapper">
        	<?php
			$urlVars = "";
			$alb = "";
            echo paginate_array($numPages,$urlVars,$alb,$currentPage);
			?>
         </div>   
    <?php
	}

} 
else //display photos in album 
{

	$album = $mainFolder.'/'.$_REQUEST['album']; 
	$files = array_diff(scandir($album), array('..', '.','thumbs'));
	$numFiles = count($files); ?>
	 
	<div class="p10-lr">
		<?php if($fullAlbum==1) { ?>
			<span class="title"><a href="<?php echo $album_page_url; ?>" class="refresh">Albums</a></span>
			<span class="title">&raquo;</span>
		<?php } ?>
		<span class="title"><?php echo $_REQUEST['album']; ?></span> - <?php echo $numFiles; ?> images
	</div>  
	   
	<div class="clear"></div>
	
	<?php
	if($numFiles == 0)
	{ ?>
	    
		 <div class="p10-lr"><p>There are no images in this album.</p></div>
	
	<?php
	}
	else	
	{			
		sort_array($files,$album,$sort_images_by_date); // rearrange array either by date or name
		$numPages = ceil( $numFiles / $numPerPage );
		
		if(isset($_REQUEST['p']))
		{
		 	$currentPage = ((int)$_REQUEST['p'] > $numPages ? $numPages : (int)$_REQUEST['p']);
		} 
		 else
		{
		 	$currentPage=1;
		}
		 			 
		$start = ($currentPage * $numPerPage) - $numPerPage;
		
		if (!is_dir($album.'/thumbs')) 
		{
			mkdir($album.'/thumbs');
			chmod($album.'/thumbs', 0777);
			//chown($album.'/thumbs', 'apache'); 
		}	 	

		for( $i=0; $i <= $numFiles; $i++ )
		{   
			if(isset($files[$i]) && is_file($album .'/'. $files[$i]))
			{   		    
				$ext = strtolower(pathinfo($files[$i], PATHINFO_EXTENSION));
				$caption = substr($files[$i], 0, -(strlen($ext)+1));
					
				if(in_array($ext, $extensions)) 
				{  				  					   
					$thumb = $album.'/thumbs/'.$files[$i];
				   
					if($i>=$start && $i<$start + $numPerPage) { 

							if (!file_exists($thumb)) {
								make_thumb($album,$files[$i],$thumb,$thumb_width); 
							}	   
 ?>
					<div class="thumb-wrapper">
						<div class="thumb">
							<a href="<?php echo $album; ?>/<?php echo $files[$i]; ?>" title="<?php echo $files[$i]; ?>" class="albumpix">
								<img src="<?php echo $thumb; ?>" alt="<?php echo $files[$i]; ?>" />
							</a>
						</div> 
						<?php if($show_captions) { ?><div class="caption"><?php echo substr($caption,0,$num_captions_chars); ?></div><?php } ?> 

					<?php 
					}

				if($i>=$start && $i<$start + $numPerPage) { ?></div><?php } }
			
			} 
			
		} ?> 
	
		<div class="clear"></div>
		  
		<div align="center" class="paginate-wrapper">
			<?php	 
			$urlVars = "album=".urlencode($_REQUEST['album'])."&amp;";
			$alb = $_REQUEST['album'];
			echo paginate_array($numPages,$urlVars,$alb,$currentPage);
			?>
		</div>
	 
	<?php	 
	} // end if numFiles not 0	 

}
?>
</div>
